apis/v1alpha1/certificate.go

```diff
@@ -0,0 +1,160 @@
+)
+
+// CertificateSpec defines the desired state of Certificate.
+type CertificateSpec struct {
+
+	// Specifies X.509 certificate information to be included in the issued certificate.
+	// An APIPassthrough or APICSRPassthrough template variant must be selected,
+	// or else this parameter is ignored. For more information about using these
+	// templates, see Understanding Certificate Templates (https://docs.aws.amazon.com/privateca/latest/userguide/UsingTemplates.html).
+	//
+	// If conflicting or duplicate certificate information is supplied during certificate
+	// issuance, Amazon Web Services Private CA applies order of operation rules
+	// (https://docs.aws.amazon.com/privateca/latest/userguide/UsingTemplates.html#template-order-of-operations)
+	// to determine what information is used.
+	APIPassthrough *APIPassthrough `json:"apiPassthrough,omitempty"`
+	// The Amazon Resource Name (ARN) that was returned when you called CreateCertificateAuthority
+	// (https://docs.aws.amazon.com/privateca/latest/APIReference/API_CreateCertificateAuthority.html).
+	// This must be of the form:
+	//
+	// arn:aws:acm-pca:region:account:certificate-authority/12345678-1234-1234-1234-123456789012
+	CertificateAuthorityARN *string                                  `json:"certificateAuthorityARN,omitempty"`
+	CertificateAuthorityRef *ackv1alpha1.AWSResourceReferenceWrapper `json:"certificateAuthorityRef,omitempty"`
+	// The certificate signing request (CSR) for the certificate you want to issue.
+	// As an example, you can use the following OpenSSL command to create the CSR
+	// and a 2048 bit RSA private key.
+	//
+	// openssl req -new -newkey rsa:2048 -days 365 -keyout private/test_cert_priv_key.pem
+	// -out csr/test_cert_.csr
+	//
+	// If you have a configuration file, you can then use the following OpenSSL
+	// command. The usr_cert block in the configuration file contains your X509
+	// version 3 extensions.
+	//
+	// openssl req -new -config openssl_rsa.cnf -extensions usr_cert -newkey rsa:2048
+	// -days 365 -keyout private/test_cert_priv_key.pem -out csr/test_cert_.csr
+	//
+	// Note: A CSR must provide either a subject name or a subject alternative name
+	// or the request will be rejected.
+	CSR    []byte                                   `json:"csr,omitempty"`
+	CSRRef *ackv1alpha1.AWSResourceReferenceWrapper `json:"csrRef,omitempty"`
+	// Alphanumeric string that can be used to distinguish between calls to the
+	// IssueCertificate action. Idempotency tokens for IssueCertificate time out
+	// after five minutes. Therefore, if you call IssueCertificate multiple times
+	// with the same idempotency token within five minutes, Amazon Web Services
+	// Private CA recognizes that you are requesting only one certificate and will
+	// issue only one. If you change the idempotency token for each call, Amazon
+	// Web Services Private CA recognizes that you are requesting multiple certificates.
+	IdempotencyToken *string `json:"idempotencyToken,omitempty"`
```

Comment:
This is not something that should be provided by the user. The controller can inject random strings when needed. Let's remove this field.

Reply:
Fixed this in commit "Remove IssueCertificateInput.IdempotencyToken"